indico/htdocs/js/utils/forms.js

```diff
@@ -64,8 +64,13 @@
         forms.find('[data-disabled-until-change]').prop('disabled', true);
         forms.each(function() {
             var $this = $(this);
-            $this.data('initialData', $this.serialize());
-            $this.data('fieldsChanged', false);
+            function _resetData() {
+                $this.data('initialData', $this.serialize());
+                $this.data('fieldsChanged', false);
+            }
+            _resetData();
```

Comment:
But we don't always need the `change` event. Only dropzones and other non-native widgets need that.